elasticdl_client/common/k8s_client.py

```diff
@@ -47,6 +56,168 @@ def append_pod_ip_to_env(env):
     return env
 
 
+def try_get_class(module, name):
+    try:
+        cls = getattr(module, name)
+        return cls
+    except AttributeError:
+        return None
+
+
+def check_list_get_type(type_name):
+    if type_name.startswith("list["):
+        return True, type_name.split("[")[1].split("]")[0]
+    else:
+        return False, None
+
+
+def get_instance_from_value(type_name, value):
+    # Args:
+    #   type_name: a class name in client, or a basic type such as string,
+    #              int, float, etc.
+    #   value: a dict corresponding to the class, or a string/int/float
+    #          value if basic type.
+    # Return: a class instance if class, the basic type value otherwise.
+
+    cls = try_get_class(client, type_name)
+    if not cls:
+        # not a class
+        is_a_list, list_type = check_list_get_type(type_name)
```

Comment:
Why should we check the list type here? Could you please give an example?

Reply:
A reference: https://github.com/kubernetes-client/python/blob/bc79976edfe01b028aef0b93bde54e002b113443/kubernetes/client/models/v1_node_selector.py#L36